src/components/Borrow/borrowUtils.tsx

```diff
@@ -0,0 +1,460 @@
+import { BN } from '@project-serum/anchor'
+import { printBN, printBNtoBN, stringToMinDecimalBN } from '@consts/utils'
+import { Decimal } from '@synthetify/sdk/lib/exchange'
+import { ActionType } from '@reducers/vault'
+interface AssetPriceData {
+  priceVal: BN
+  assetScale: number
+  symbol: string | null
+  maxAvailable: BN
+  balance: BN
+}
+export const calculateAmountCollateral = (
+  assetTo: AssetPriceData,
+  assetFrom: AssetPriceData,
+  amount: BN,
+  cRatio: string
+) => {
+  const cRatioBN = stringToMinDecimalBN(cRatio)
+  const amountBeforeCalculations = assetTo.priceVal
+    .mul(amount)
+    .div(assetFrom.priceVal)
```

Comment:
i think you should move divs to end of operations' chains, if only there will be no risk of overflow or sth like that